shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/presentation/viewmodel/handler/DefaultWillFeedHandler.kt

```diff
@@ -0,0 +1,25 @@
+package com.epmedu.animeal.feeding.presentation.viewmodel.handler
+
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.StateDelegate
+import com.epmedu.animeal.feeding.presentation.event.WillFeedEvent
+import com.epmedu.animeal.feeding.presentation.viewmodel.WillFeedState
+
+class DefaultWillFeedHandler(
+    stateDelegate: StateDelegate<WillFeedState>
+) : WillFeedHandler, StateDelegate<WillFeedState> by stateDelegate {
+
+    override fun handleWillFeedEvent(event: WillFeedEvent) {
+        when (event) {
+            WillFeedEvent.DismissWillFeedDialog -> dismissWillFeedDialog()
+            WillFeedEvent.ShowWillFeedDialog -> showWillFeedDialog()
+        }
+    }
+
+    private fun showWillFeedDialog() {
+        updateState { WillFeedState.Showing }
```

Comment:
It's not working this way, because state of screens don't receive this update.

Reply:
Fixed, willFeedState was updated at handler only but never passed to view model. Will update more on PR description